parser/js/README.md

```diff
@@ -0,0 +1,145 @@
+# HMDA File Parser
+
+Standalone parser for the Home Mortgage Disclosure Act (HMDA) submission file, as per the 2017 [File Specification Guide](http://www.consumerfinance.gov/data-research/hmda/static/for-filers/2017/2017-HMDA-FIG.pdf)
+
+## Building
+
+The standalone HMDA file parser is built with [Scala.js](https://www.scala-js.org/).
+To build this project, clone the repository from GitHub:
+
+```shell
+git clone https://github.com/cfpb/hmda-platform.git
+```
+
+[`SBT`](https://github.com/sbt/sbt) is required to build the library.
+
+```shell
+cd hmda-platform
+$ sbt
+> project parserJS
+> fullOptJS
+```
+
+This will build the production ready `JavaScript` library containing the parser, and place it in parser/js/target/scala-2.12/parserjs-opt.js
+
+NOTE: Building the project is only necessary if you intend to work with it locally from source code, or work on the implementation.
+If you just want to use the library, see the next section
+
+## Using
+
+This library is published to [NPM](https://www.npmjs.com/), to run in a Node.js environment, do the following:
+
+```javascript
+npm install hmda-file-parser
+```
+
+This will install the latest version of the library. The parser accepts `Transmittal Sheet` or `Loan Application Register` strings as input data. 
+
+To instantiate the parser:
+
+```javascript
+var js = require('hmda-file-parser');
+var parser = new js.hmda.parser.fi.CsvParser();
+```
+Here are a couple of examples parsing HMDA data
+
+* Transmittal Sheet
+
+```javascript
+var ts = parser.parseTs('1|21|1|201503111034|2017|35-0704860|10|CENTRAL FIRST BANK|221 Hesburgh Library|Notre Dame|IN|46556|FIRST BANK|2566 S. Kinnickinnic Ave.|Milwaukee|WI|53207|Anne Shirley|574-555-2000|574-555-2001|as@centerfirst.COM');
```

Comment:
Can we alter it a bit?  We've been caught once by the assumption that data elsewhere has already been checked.  It would be safer to have data that is obviously not realistic.